src/devTools.js

```diff
@@ -50,13 +60,41 @@ function connectToServer(options) {
   watch();
 }
 
-export function start(options) {
+
+// obtains hostname from the promise and then performs the callback
+// also, overwrites `optionsObject.hostname`
+// obtained hostname is stored in `obtainedHostname` variable
+// if it's already obtained, skip fetching
+async function obtainHostname(hostnamePromise, optionsObject, callback) {
+  if (optionsObject.hostname) {
+    // hostname is in options
+    callback();
+  } else if (obtainedHostname) {
+    // hostname has been obtained using the promise already
+    optionsObject.hostname = obtainedHostname;
+    callback();
+    return;
+  }
+  // hostname needs to be obtained
+  try {
+    obtainedHostname = await hostnamePromise;
+    optionsObject.hostname = obtainedHostname;
+    callback();
+  } catch (err) {
+    console.log('Error obtaining socket hostname: ' + err.toString());
+  }
+}
```

Comment:
If you take my advice above, you can rewrite this code to be way simpler:

```lang=js
if (optionsObject.hostname) {
  return;
}
try {
  optionsObject.hostname = await hostnamePromise;
} catch (err) {
  console.log('Error obtaining socket hostname: ' + err.toString());
  // throw?
}
```

Note that I got rid of the `callback`... that can simply be handled by the caller after `await`ing `obtainHostname`.

Reply:
please see [my reply](https://github.com/karol-bisztyga/remotedev/pull/1/files#r607664267)